Libs/GObject/Core/Classes/Object.TypeDictionary.cs

```diff
@@ -0,0 +1,153 @@
+                // Use GetGType function for reverse mapping.
+
+                // Search all System.Type which contain a [Wrapper(TypeName)]
+                // for the corresponding type.
+
+                // Possible Idea: Autogenerate a 'RegisterTypes.cs' file that
+                // on startup will add every type to the type dictionary?
+
+                // Quick Path: Find the first 'Word' in the type and lookup
+                // assemblies by that name. Do we hardcode references to 'Pango',
+                // 'Gtk', etc?
+
+                // foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+                // {
+                //     assembly.GetType()
+                // }
+
+                // Search through unloaded assemblies?
+
+                // TODO: For now, we'll just look through the type's inheritance chain
+                // and find the first already registered type (e.g. GtkWidget). Effectively,
+                // the lowest-common-denominator of functionality will be exposed.
+                while (!Contains(gtype))
+                {
+                    ulong parent = Sys.Methods.type_parent(gtype.Value);
+                    if (parent == 0)
+                        throw new Exception("Could not get Type from GType");
+
+                    // TODO: One-way registration?
+                    
+                    gtype = new Sys.Type(parent);
+                }
+
+                return gtypedict[gtype];
+            }
+
+            // Get GType from System.Type
+            internal static Sys.Type Get(System.Type type)
+            {
+                // Check Type Dictionary
+                if (typedict.TryGetValue(type, out var gtype))
+                    return gtype;
+
+                // If we are looking up a type that is not yet in
+                // the type dictionary, we are most likely registering
+                // a new type. Therefore, we should register the type
+                // and parent types recursively now to avoid having to
+                // do this in the future.
+                
+                // Retrieve the GType accordingly
+                if (IsSubclass(type))
+                {
+                    // We are a subclass
+                    // RegisterNativeType will recursively add this
+                    // and all parent types to the type dictionary
+                    RegisterNativeType(type);
+                    return typedict[type];
+                }
+                
+                // We are a wrapper, so register types recursively
+                Console.WriteLine("Registering Recursively");
+                System.Type baseType = type;
+                while (!Contains(baseType))
+                {
+                    Console.WriteLine(baseType.Name);
+                    var methodInfo = GetGTypeMethodInfo(baseType)!;
+                    var typeid = (ulong) methodInfo.Invoke(null, null);
```

Comment:
Got an exception here:

```log
System.InvalidCastException: Unable to cast object of type 'GObject.Type' to type 'System.UInt64'.
```

Fixed with:

```suggestion
                    var typeid = (ulong) (Type) methodInfo.Invoke(null, null);
```

After added an explicit cast to `ulong` in the `GObject.Type` struct.

Reply:
I remember changing this at one point, not sure why it's still like this. You're right 👍

Reply:
Resolved in 0148472